cocos/core/animation/target-path.ts

```diff
@@ -71,3 +71,28 @@ export class ComponentPath implements ICustomTargetPath {
         return result;
     }
 }
+
+/**
+ * Evaluate a path from specified root.
+ * @param root The root object.
+ * @param path The path.
+ */
+export function evaluatePath(root: any, ...path: TargetPath[]) {
+    let result = root;
+    for (const segment of path) {
+        if (isPropertyPath(segment)) {
```

Comment:
Because we want to support array accessing in same way. For instance, paths like `[new animation.ComponentPath('ModelComponent'), 'materials', 2]` indicates "The 3rd element of `materials` property(which is an array) of component `ModelComponent`.